chore: revisit proxy's worker configuration

SantiagoPittella opened the pull request:
closes #1366

sergerad reviewed (approved):
LGTM TY.

bobbinth commented on bin/proving-service/README.md:
````diff
@@ -105,16 +105,22 @@ miden-proving-service start-worker
 
 ```bash
-miden-proving-service start-proxy --prover-type transaction [worker1] [worker2] ... [workerN]
+miden-proving-service start-proxy --prover-type transaction [worker1],[worker2],...,[workerN]
````
Given that the list of workers is optional, it feels like an option-based approach (i.e., with `--workers` flag) could be more appropriate here - e.g.,:
```
miden-proving-service start-proxy --prover-type transaction --workers [worker1],[worker2],...,[workerN]
```

bobbinth reviewed (approved):
Looks good! Thank you! I left two small comments inline.

bobbinth commented on bin/proving-service/src/proxy/worker.rs:
```diff
@@ -32,7 +32,7 @@ const MPS_PROXY_VERSION: &str = env!("CARGO_PKG_VERSION");
 #[derive(Debug, Clone)]
 pub struct Worker {
     backend: Backend,
-    status_client: StatusApiClient<Channel>,
+    status_client: Option<StatusApiClient<Channel>>,
     is_available: bool,
     health_status: WorkerHealthStatus,
     version: String,
```
Not from this PR, but it would be good to add comments for the fields to explain what they are - especially for `is_available` and `health_status` fields (and the description on lines 30 - 31 also needs to be updated).

bobbinth commented on bin/proving-service/src/proxy/worker.rs:
```diff
@@ -104,25 +121,52 @@ impl Worker {
             return;
         }
 
+        // If we don't have a status client, try to recreate it
+        if self.status_client.is_none() {
+            let failed_attempts = self.num_failures();
+            self.set_health_status(WorkerHealthStatus::Unhealthy {
+                num_failed_attempts: failed_attempts + 1,
+                first_fail_timestamp: match &self.health_status {
+                    WorkerHealthStatus::Unhealthy { first_fail_timestamp, .. } => {
+                        *first_fail_timestamp
+                    },
+                    _ => Instant::now(),
+                },
+                reason: "No status client available".to_string(),
+            });
+            return;
+        }
```
I may be missing something, but how does this recreate the status client?

bobbinth commented on bin/proving-service/src/proxy/worker.rs:
```diff
@@ -104,25 +121,52 @@ impl Worker {
 
+        // If we don't have a status client, try to recreate it
+        if self.status_client.is_none() {
+            let failed_attempts = self.num_failures();
+            self.set_health_status(WorkerHealthStatus::Unhealthy {
+                num_failed_attempts: failed_attempts + 1,
+                first_fail_timestamp: match &self.health_status {
+                    WorkerHealthStatus::Unhealthy { first_fail_timestamp, .. } => {
+                        *first_fail_timestamp
+                    },
+                    _ => Instant::now(),
+                },
+                reason: "No status client available".to_string(),
+            });
```
This code (and the code on lines 145 - 156 below) is a bit difficult to follow. Maybe it would make sense to create a small helper method - something like `set_unhealthy_status()`?

bobbinth commented on bin/proving-service/src/proxy/worker.rs:
```diff
@@ -32,7 +32,7 @@ const MPS_PROXY_VERSION: &str = env!("CARGO_PKG_VERSION");
 pub struct Worker {
     backend: Backend,
-    status_client: StatusApiClient<Channel>,
+    status_client: Option<StatusApiClient<Channel>>,
```
Would it make sense to create a simple wrapper struct (e.g., `StatusClient`) which would handle status checking (and reconnecting) logic internally? It could look something like this:
```rust
struct StatusClient(Option<StatusApiClient<Channel>>);

impl StatusClient {

    /// Same as current create_status_client()
    pub async fn new(
        address: String,
        connection_timeout: Duration,
        total_timeout: Duration,
    ) -> Result<Self, ProvingServiceError> {
        ...
    }

    /// Handles status fetching and reconnect logic.
    pub async fn check_status(&self, current_status: WorkerHealthStatus) -> WorkerHealthStatus {

    }
}
```

bobbinth reviewed (commented):
Thank you! Looks good. I left some comments inline. I think most of them are related to a PR that was merged into this PR.

SantiagoPittella commented:
@bobbinth already addressed your comments. Regarding the last couple ones, I'm working on refactoring a bit the status check part, it grown too much and doesn't have proper error handling. Will send another PR to tackle it.

bobbinth commented:
I think this is fine to merge - but the question is do we do it now here or do we move this (or maybe #429) to `miden-node` once we move the proving service there.

cc @sergerad.

sergerad commented:
> I think this is fine to merge - but the question is do we do it now here or do we move this (or maybe #429) to `miden-node` once we move the proving service there.
> 
> cc @sergerad.

Go ahead and merge this 👍
